src/Metable.php

```diff
@@ -39,9 +39,19 @@ trait Metable
     public static function bootMetable()
     {
         // delete all attached meta on deletion
-        static::deleted(function (self $model) {
-            $model->purgeMeta();
-        });
+        
+        if (!in_array('Illuminate\Database\Eloquent\SoftDeletes', class_uses(self::class)))
+        {
+            static::deleted(function(self $model) {
+                $model->purgeMeta();
+            });
+        }
+        else
+        {
+            static::forceDeleted(function(self $model) {
+                $model->purgeMeta();
+            });
+        }
```

Comment:
Styling nits:
- see the failing CI for PSR code style
- use `::class` to refer to that trait
- should you be using `get_called_class()` instead of `self::class`, in case of inheritance (e.g. A uses Metable, B extends A uses SoftDeletes)